### PR TITLE
Add a new cleaner bot

### PR DESCRIPTION
**Pull request by EmilyBourne**
This PR contains the new bot. The bot will not work until it is merged to the default branch. Testing is carried out at EmilyBourne/pyccel

### CODE REVIEW

File: ci_tools/bot_messages/welcome_friend.txt [hunk 1]
```diff
@@ -1,5 +1,3 @@
 Hello again! Thank you for this new pull request :star_struck:.
 
-Don't forget to let me know when it is complete with the command `/bot mark as ready`.
-
-Here is your checklist:
+Here is your checklist. Please tick items of when you have completed them or determined that they are not necessary for this pull request:
```
Comment (yguclu) [inline ci_tools/bot_messages/welcome_friend.txt:3]: Typo: of -> off

File: ci_tools/check_python_capitalisation.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+""" Look for instances of python which are not capitalised and report errors
```
Comment (yguclu) [inline ci_tools/check_python_capitalisation.py:1]: LOL

**Review by yguclu — CHANGES_REQUESTED** [review]

File: docs/compiler.md [hunk 3]
```diff
@@ -64,4 +64,4 @@ To avoid these problems, Pyccel ignores Conda paths when searching for compilers
 
 By using the expected compiler, the expected libraries are correctly linked.
 
-Pyccel offers ``--conda-warnings`` flag with options ``off``, ``basic``, and ``verbose`` to control the visibility of Conda-related warnings.
\ No newline at end of file
+Pyccel offers ``--conda-warnings`` flag with options ``off``, ``basic``, and ``verbose`` to control the visibility of Conda-related warnings.
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply docs/compiler.md]: Added more description to docs : 05aa3698

File: ci_tools/check_python_capitalisation.py (hunk 2 above)
Comment (EmilyBourne) [reply ci_tools/check_python_capitalisation.py]: Fixed [b1f9792](https://github.com/pyccel/pyccel/pull/1449/commits/b1f9792642c2c39f09d05645a3983bd43553229f)

File: ci_tools/bot_messages/welcome_newcomer.txt [hunk 4]
```diff
@@ -2,7 +2,7 @@ Hello! Welcome to Pyccel! Thank you very much for your contribution :heart:.
...
+To begin with I will give you a short checklist to make sure your pull request is complete. Please tick items of when you have completed them or determined that they are not necessary for this pull request. If you want me to run any specific tests to check out corner cases that you can't easily check on your computer, you can request this using the command `/bot run X`. Use the command `/bot show tests` to see a full list of the tests I can run. Once you have finished preparing your pull request and are ready to request reviews just take your PR out of draft, or let me know with the command `/bot mark as ready`. I will then run the full suite of tests to check that everything is as neat as you think before asking other contributors for reviews. You can get a full list of commands that I understand using `/bot commands`.
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply ci_tools/bot_messages/welcome_newcomer.txt]: Fixed [0583484](https://github.com/pyccel/pyccel/pull/1449/commits/05834841d9ac63ca53ddfc17a44bc7ee453efbd2)

File: ci_tools/bot_messages/welcome_friend.txt (hunk 1 above)
Comment (EmilyBourne) [reply ci_tools/bot_messages/welcome_friend.txt]: Fixed [0583484](https://github.com/pyccel/pyccel/pull/1449/commits/05834841d9ac63ca53ddfc17a44bc7ee453efbd2)

File: .github/actions/generate_bot_token/action.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+name: "Generate bot token"
+inputs:
+  bot_pem:
+    description: "The private key which identfies the bot"
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply .github/actions/generate_bot_token/action.yml]: Fixed : [aa87910](https://github.com/pyccel/pyccel/pull/1449/commits/aa87910e61c31be72319a7622821bc25d6623444)

**Review by yguclu — APPROVED** [review]
Looks good to me. Let's see if everything works now 😅
